src/test/compile-fail/bad-mid-path-type-params.rs

```diff
@@ -39,12 +39,14 @@ impl Trait<isize> for S2 {
 fn foo<'a>() {
     let _ = S::new::<isize,f64>(1, 1.0);
     //~^ ERROR too many type parameters provided
+    //~| ERROR mismatched types
```

Comment:
It's inconsistent between type and value paths, although I think that *after* the PR, `check_path_parameter_count` could take the segment mutably and set it to `None`, which *should* get the old behavior of injecting inference variables.